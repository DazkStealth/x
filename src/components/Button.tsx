import styled from 'styled-components';

const Button = styled.button`
  font-size: 18px;
  font-weight: 700;
  padding: 15px 45px;
  border: none;
  border-radius: 8px;
  background: ${({ theme }) => theme.color.background.buttonPrimary};
  color: ${({ theme }) => theme.color.text.buttonPrimary};
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  margin-bottom: 15px;

  &:hover {
    opacity: 0.7;
  }

  &:active {
    opacity: 0.4;
  }

  &:focus {
    outline: none;
  }
`;

export default Button;
