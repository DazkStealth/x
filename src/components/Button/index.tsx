import styled from 'styled-components';

const Button = styled.button<{ fullWidth?: boolean }>`
  font-size: 18px;
  font-weight: 700;
  padding: 15px 45px;
  border: none;
  border-radius: 100px;
  background: ${({ theme }) => theme.color.background.buttonPrimary};
  color: ${({ theme }) => theme.color.text.buttonPrimary};
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  margin-bottom: 15px;
  
  ${({ fullWidth }) => fullWidth && `
    width: 100%;
  `}

  &:hover {
    opacity: 0.7;
  }

  &:active {
    opacity: 0.4;
  }

  &:focus {
    outline: none;
  }
`;

export default Button;
