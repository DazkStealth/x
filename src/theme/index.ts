export interface Theme {
  font: {
    primary: string;
  }
  color: {
    background: {
      body: string;
      bottomMenu: string;
      bottomMenuItem: string;
      loadingLogo: string;
      buttonPrimary: string;
    },
    text: {
      body: string;
      bottomMenuItem: string;
      bottomMenuItemActive: string;
      loadingLogo: string;
      buttonPrimary: string;
    },
    border: {
      bottomMenuItemBottomActive: string;
    }
  }
}

export const defaultTheme: Theme = {
  font: {
    primary: '"Roboto", sans',
  },
  color: {
    background: {
      body: '#24142f',
      bottomMenu: 'rgba(50,50,66,0.7)',
      bottomMenuItem: '#444d55',
      loadingLogo: '#fff',
      buttonPrimary: '#c63bea',
    },
    text: {
      body: '#fff',
      bottomMenuItem: '#fff',
      bottomMenuItemActive: '#444d55',
      loadingLogo: '#997cfa',
      buttonPrimary: '#fff',
    },
    border: {
      bottomMenuItemBottomActive: '#fff',
    }
  },
};
