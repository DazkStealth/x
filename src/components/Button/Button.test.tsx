import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// components
import Button from './';

// theme
import { defaultTheme } from '../../theme';

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Button>Some regular text.</Button>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

    const treeElement = tree as ReactTestRendererJSON;
    expect(treeElement.children?.length).toBe(1);
    expect(treeElement.children?.[0]).toBe('Some regular text.');
    expect(treeElement.type).toBe('button');
    expect(treeElement).toHaveStyleRule('color', defaultTheme.color.text.buttonPrimary);
  });
});
