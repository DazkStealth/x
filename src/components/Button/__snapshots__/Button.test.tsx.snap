// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
.c0 {
  font-size: 18px;
  font-weight: 700;
  padding: 15px 45px;
  border: none;
  border-radius: 8px;
  background: #c63bea;
  color: #fff;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  margin-bottom: 15px;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 0.4;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
>
  Some regular text.
</button>
`;
